sysl2/sysl/seqs.go

```diff
@@ -0,0 +1,1081 @@
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/anz-bank/sysl/src/proto"
+	"github.com/sirupsen/logrus"
+)
+
+type indentWriter struct {
+	bytes.Buffer
+	ind      int
+	complete bool // if the current position is a complete line
+}
+
+func (w *indentWriter) Write(p []byte) (n int, err error) {
```

Comment:
I haven't analysed this in depth, so I could be wrong, but my gut-feel is that this function is more complicated than necessary.